prov/gni/src/gnix_cm_nic.c

```diff
@@ -464,7 +489,7 @@ int _gnix_cm_nic_enable(struct gnix_cm_nic *cm_nic)
 	if (cm_nic == NULL)
 		return -FI_EINVAL;
 
-	fabric = cm_nic->fabric;
+	fabric = cm_nic->domain->fabric;
```

Comment:
While we're on the topic of asserts, seems like we should check the domain is *not* null here, before dereferencing it to get fabric.